src/core/common/settings.hpp

```diff
@@ -152,6 +151,7 @@ class SettingsBase : public InstanceLocator
         kKeySlaacIidSecretKey = 0x0007, ///< Secret key used by SLAAC module for generating semantically opaque IID
     };
 
+protected:
```

Comment:
is this change necessary?

Reply:
This is needed as radio_spinel is not inherited from settings